core/node/state_keeper/src/keeper.rs

```diff
@@ -670,23 +671,23 @@ impl ZkSyncStateKeeper {
                 reason: Halt::NotEnoughGasProvided,
             } => {
                 let error_message = match &exec_result {
-                    TxExecutionResult::BootloaderOutOfGasForTx => "bootloader_tx_out_of_gas",
+                    TxExecutionResult::BootloaderOutOfGasForTx => ErrorMessage::BootloaderOutOfGas,
                     TxExecutionResult::RejectedByVm {
                         reason: Halt::NotEnoughGasProvided,
-                    } => "not_enough_gas_provided_to_start_tx",
+                    } => ErrorMessage::NotEnoughGasProvided,
                     _ => unreachable!(),
                 };
+                let criterion = error_message.as_static_str();
+
                 let resolution = if is_first_tx {
-                    SealResolution::Unexecutable(error_message.to_string())
+                    error_message.into()
                 } else {
                     SealResolution::ExcludeAndSeal
                 };
-                AGGREGATION_METRICS.inc(error_message, &resolution);
+                AGGREGATION_METRICS.inc(criterion, &resolution);
```

Comment:
Nit: Here as well, from the encapsulation standpoint it would make sense to refactor `inc()` to accept `&ErrorMessage` (or `&UnexecutableReason`; see the other comment), so that mapping to labels is encapsulated in the `metrics` module.

Reply:
@slowli , if I will incapsulate mapping, then I could not get `ErrorMessage` that is needed to create `resolution`, that has to be return.

Reply:
I think I understand what the problem is. `TxAggregationMetrics::inc()` is used in two places with somewhat different semantics for the criterion argument. I think it's definitely inappropriate to use a human-readable label produced by `as_static_str()` here; but even if you make `error_message_as_metric_label` a method in `ErrorMessage`, it would still produce non-backward compatible labels. (Is backward compatibility a concern?) IDK, it may make sense to initialize `criterion` manually together with `error_message` in the `match` above.